solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -0,0 +1,1362 @@
+      * @param _targetId    target reserve id
+    */
+    function readReserveBalances(uint256 _sourceId, uint256 _targetId) internal view returns (uint256, uint256) {
+        require(_sourceId + _targetId == 3, "ERR_INVALID_RESERVES");
+        uint256 reserveBalancesLocal = __reserveBalances;
+        return (
+            (reserveBalancesLocal >> ((_sourceId - 1) * 128)) & MAX_UINT128,
+            (reserveBalancesLocal >> ((_targetId - 1) * 128)) & MAX_UINT128
+        );        
+    }
+
+    /**
+      * @dev writes the stored reserve balance for a given reserve id
+      *
+      * @param _reserveId       reserve id
+      * @param _reserveBalance  reserve balance
+    */
+    function writeReserveBalance(uint256 _reserveId, uint256 _reserveBalance) internal {
+        require(_reserveBalance <= MAX_UINT128, "ERR_RESERVE_BALANCE_OVERFLOW"); 
+        uint256 otherBalance = readReserveBalance(3 - _reserveId);
+        __reserveBalances = (_reserveBalance << ((_reserveId - 1) * 128)) | (otherBalance << ((2 - _reserveId) * 128));
+    }
+
+    /**
+      * @dev writes the stored reserve balances
+      *
+      * @param _sourceId        source reserve id
+      * @param _targetId        target reserve id
+      * @param _sourceBalance   source reserve balance
+      * @param _targetBalance   target reserve balance
+    */
+    function writeReserveBalances(uint256 _sourceId, uint256 _targetId, uint256 _sourceBalance, uint256 _targetBalance) internal {
+        require(_sourceBalance <= MAX_UINT128 && _targetBalance <= MAX_UINT128, "ERR_RESERVE_BALANCE_OVERFLOW"); 
+        __reserveBalances = (_sourceBalance << ((_sourceId - 1) * 128)) | (_targetBalance << ((_targetId - 1) * 128));
+    }
+
+    /**
+      * @dev syncs the stored reserve balance for a given reserve with the real reserve balance
+      *
+      * @param _reserveToken    address of the reserve token
+    */
+    function syncReserveBalance(IERC20Token _reserveToken) internal {
+        uint256 reserveId = __reserveIds[_reserveToken];
+        uint256 balance = _reserveToken == ETH_RESERVE_ADDRESS ? address(this).balance : _reserveToken.balanceOf(address(this));
+        writeReserveBalance(reserveId, balance);
+    }
+
+    /**
+      * @dev syncs all stored reserve balances
+    */
+    function syncReserveBalances() internal {
+        IERC20Token _reserveToken0 = __reserveTokens[0];
+        IERC20Token _reserveToken1 = __reserveTokens[1];
+        uint256 balance0 = _reserveToken0 == ETH_RESERVE_ADDRESS ? address(this).balance : _reserveToken0.balanceOf(address(this));
+        uint256 balance1 = _reserveToken1 == ETH_RESERVE_ADDRESS ? address(this).balance : _reserveToken1.balanceOf(address(this));
+        writeReserveBalances(1, 2, balance0, balance1);
+    }
+
+    /**
+      * @dev syncs all stored reserve balances, excluding the ether value of the current transaction
+    */
+    function syncReserveBalancesExcludingCurrentValue() internal {
```

Comment:
Where exactly is `msg.sender` being used here?
You probably mean `msg.value`; I thought about passing it from the outside to function `syncReserveBalances`, but it would yield an unnecessary cost of this function, which mostly doesn't need to receive it (and additional code for passing to it `0` everywhere in the code).
So your suggestion is to pass `msg.value` to this function from the outside?
Can do, but any reason why you insist on this?

Reply:
yup, for the reason I wrote above, high risk to pass msg.sender/value to internal functions, can lead to unexpected results later on as the caller has no idea that the function actually relies on external data.

Reply:
Changed to input parameter.

Reply:
the function name/docs should be corrected, as a matter of fact, I should simply override the default `syncReserveBalances` function with another one that receives another argument which represents the ETH amount to ignore.

Reply:
Renamed `syncReserveBalancesExcludingCurrentValue` to `syncReserveBalances`.